src/avro.erl

```diff
@@ -309,6 +310,143 @@ decode(avro_json, JSON, TypeOrName, StoreOrLkup, Hook) ->
 decode(avro_binary, Bin, TypeOrName, StoreOrLkup, Hook) ->
   avro_binary_decoder:decode(Bin, TypeOrName, StoreOrLkup, Hook).
 
+%% @doc Checks whether two schemas are compatible in sense that an event
+%% encoded with the writer schema can be decoded by the reader schema.
+%% Schema compatibility is defined in
+%% https://avro.apache.org/docs/1.8.1/spec.html#Schema+Resolution
```

Comment:
Please help to document the compatibility check rules.
At least the parts that are different from spec.
i.e. spec is about decoded object compatibility to reader schema.
but we check reader / writer schema compatibility directly.